src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,959 @@
+        /// For example:
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.Below): below 100
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.Above): above 300
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.Out): out-of-window
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.In): in window
+        /// </summary>
+        /// <param name="threshLow">
+        /// The threshold under which the device raises an interrupt if detectionMode =
+        /// WindowDetectionMode.Below
+        /// </param>
+        /// <param name="threshHigh">
+        /// The threshold above which the device raises an interrupt if detectionMode =
+        /// WindowDetectionMode.Above
+        /// </param>
+        /// <param name="detectionMode">The <see cref="WindowDetectionMode" /> where 0 = below, 1 = above, 2 = out, and 3 = in</param>
+        public void SetDistanceThreshold(ushort threshLow, ushort threshHigh, WindowDetectionMode detectionMode)
+        {
+            byte temp = ReadByte((ushort)Registers.SYSTEM__INTERRUPT_CONFIG_GPIO);
+            temp &= 0x47;
+            WriteRegister((ushort)Registers.SYSTEM__INTERRUPT_CONFIG_GPIO,
+                (byte)(temp | ((byte)detectionMode & 0x07) | 0x40));
+            WriteUInt16((ushort)Registers.SYSTEM__THRESH_HIGH, threshHigh);
+            WriteUInt16((ushort)Registers.SYSTEM__THRESH_LOW, threshLow);
+        }
+
+        /// <summary>
+        /// Returns the <see cref="WindowDetectionMode" />.
+        /// </summary>
+        public WindowDetectionMode DistanceThresholdWindowDetectionMode
+        {
+            get
+            {
+                byte temp = ReadByte((ushort)Registers.SYSTEM__INTERRUPT_CONFIG_GPIO);
+                return
+                    (WindowDetectionMode)(temp &
+                                          0x7);
+            }
+        }
+
+        /// <summary>
+        /// Returns the low threshold in mm.
+        /// </summary>
+        public ushort DistanceThresholdLow
+        {
+            get
+            {
+                return ReadUInt16((ushort)Registers.SYSTEM__THRESH_LOW);
+            }
+        }
+
+        /// <summary>
+        /// Returns the high threshold in mm.
+        /// </summary>
+        public ushort DistanceThresholdHigh
+        {
+            get
+            {
+                return ReadUInt16((ushort)Registers.SYSTEM__THRESH_HIGH);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="Roi" />, the position of which is centered about the optical center.
+        /// The smallest acceptable ROI size is 4.
+        /// </summary>
+        public Roi Roi
+        {
+            get
+            {
+                byte temp = ReadByte((ushort)Registers.ROI_CONFIG__USER_ROI_REQUESTED_GLOBAL_XY_SIZE);
+
+                return new Roi((ushort)((temp & 0x0F) + 1), (ushort)(((temp & 0xF0) >> 4) + 1));
+            }
+            set
```

Comment:
nit, you need an extra CR between } and set.

Reply:
Done